scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/pskstore/AdvancedPskStore.java

```diff
@@ -0,0 +1,136 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Bosch.IO GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v20.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch.IO GmbH - initial creation
+ *                    Inspired by the MasterSecretDeriver contribute
+ *                    Jaimie Whiteside (Arm). 
+ *                    Obsoletes that MasterSecretDeriver.
+ ******************************************************************************/
+package org.eclipse.californium.scandium.dtls.pskstore;
+
+import java.net.InetSocketAddress;
+
+import javax.crypto.SecretKey;
+
+import org.eclipse.californium.scandium.DTLSConnector;
+import org.eclipse.californium.scandium.dtls.ConnectionId;
+import org.eclipse.californium.scandium.dtls.PskSecretResult;
+import org.eclipse.californium.scandium.dtls.PskSecretResultHandler;
+import org.eclipse.californium.scandium.dtls.PskPublicInformation;
+import org.eclipse.californium.scandium.util.ServerNames;
+
+/**
+ * New advanded PSK store with support for asynchrounous PSK store and HSM.
+ * 
+ * Returns psk secret result instead of PSK's secret key. The secret must either
+ * be a master secret (algorithm "MAC"), or a PSK secret key (algorithm "PSK").
+ * If required, the psk secret result maybe returned asynchrounos using a
```

Comment:
My English isn't the best, indeed ;-). Anyway, if "returned" is the verb, "maybe" is the adverb [maybe-or-may-be](https://dictionary.cambridge.org/de/grammatik/britisch-grammatik/maybe-or-may-be).

@sophokles73 
"maybe" or "may be" you can help us with your English?

Reply:
To me : 
"maybe" = "perhaps"
and
"may be" = "can/could be"

Reply:
FMPOV it should read
> If required, the PSK secret result may be returned asynchronously using a ...

However, *if required* is quite vague so I would prefer a concrete example for when this would be desirable/feasible. WDYT?